src/kOS.Safe/Execution/SurboutineContext.cs

```diff
@@ -1,39 +1,8 @@
 namespace kOS.Safe.Execution
 {
     /// <summary>
-    /// <p>The context record that gets pushed onto the stack to store what you need to
+    /// The context record that gets pushed onto the stack to store what you need to
     /// know to return from a subroutine.
-    /// </p>
-    /// 
-    /// <p>At the moment it only contains the instruction pointer to return to.</p>
-    /// 
-    /// <p>Why isn't it just an integer then?</p>
-    /// 
-    /// <p>It was just an integer before, but that makes it impossible to tell the
-    /// difference between an integer expression value on the stack and the
-    /// function's return point.  Having a special type for storing the return
-    /// pointer effectively puts section separators into the stack - everything
-    /// in between two SubroutineContext records on the stack belongs to that same
-    /// nesting level of function call.</p>
-    /// 
-    /// <p>It can also be expanded to include more things than the return IP.  For
-    /// example if we ever want to implement local variables, it would make sense
-    /// to store a reference to the Dictionary namespace of them in this context record, so it
-    /// too can be popped back when returning from the routine.</p>
-    /// 
-    /// <p>It is also useful if we want to implement a premature abort from a routine
-    /// (i.e. exit or return statement).  With a special named class like this on
-    /// the stack at the point where the current subroutine was begun, it will be
-    /// possible to identify how much of the top of the stack to erease if we want to
-    /// abort prematurely and return (everything on top of the topmost SubroutineContext
-    /// is the stuff that was added by the current routine).</p>
-    /// 
-    /// <p>It also makes it possible to implement varying arguments passed to a function
-    /// this way, because we can find out how many of the objects on top of the stack are
-    /// the parameters to the current function - When the function starts everything in
-    /// the stack above where the SubroutineContext record is will be a parameter.  (Also,
-    /// it should be possible to detect the wrong number of arguments better this way, for
-    /// the same reason).</p>
```

Comment:
I deleted most of this text purely because it was old and talking about "future plans" that have in fact already been implemented.